### DISCUSSION TITLE
Assign user roles programatically

### DISCUSSION THREAD

**Discussion by rytina** [structurizr/onpremises#96] [General]
Hi,
is it possible to assign access rights for a workspace to a user programmatically with the Structurizr Java client library, CLI or the REST-API?

Regards,
Andy

**Comment by simonbrowndotje** [comment]
How are you creating your workspace?

**Reply by rytina** [reply]
Hi,
I create workspaces with the Admin API: curl -X POST --header "X-Authorization: ****" http://localhost:8080/api/workspace

**Reply by simonbrowndotje** [reply]
Sorry, I mean how are you authoring your workspace? Java client, DSL, etc?

**Reply by rytina** [reply]
Ah, I create them via DSL.

**Reply by simonbrowndotje** [reply]
In that case, just add them as users in the DSL itself -> https://docs.structurizr.com/dsl/language#users

Edit1 : example -> https://github.com/structurizr/dsl/blob/master/src/test/dsl/test.dsl#L335
Edit 2: Oh, I finished porting the docs over a few days ago, and this is now included in the docs -> https://docs.structurizr.com/onpremises/role-based-access#configuring-role-based-access-via-the-web-api

**Reply by rytina** [reply]
Amazing thanks, permissions as code as well 👍
